### ISSUE TITLE
Suggestion: cluster-robust standard errors based on sandwich instead of clubSandwich

### ISSUE THREAD

**Issue by awcm0n** [strengejacke/ggeffects#198]
When calculating cluster-robust standard errors for large samples, clubSandwich produces memory allocation errors but sandwich does not in my experience. Since cluster-robust estimators have recently become available in the sandwich package, perhaps you could provide the option to use those estimators?

**Comment by strengejacke** [comment]
I'm not sure, but I think you can already use the cluster-robust estimators from the _sandwich_ package. Which vcov-function is doing that? `vcovCL()`?

**Comment by awcm0n** [comment]
Yes, I just realized that `vcovCL()` from the _sandwich_ package works. Unfortunately, I'm also encountering a memory allocation problem, even though I can compute the the same cluster-robust estimate of the covariance matrix with `vcovCL()` outside of ggpredict. Is there a way to just plug in the estimate into ggpredict?

**Comment by strengejacke** [comment]
You can try to call `ggpredict()` w/o robust SE, and then call [`vcov()`](https://strengejacke.github.io/ggeffects/reference/vcov.html) on the returned ggeffects-object (where you specify the desired robust estimation type).

**Comment by awcm0n** [comment]
Unfortunately, I'm still getting the memory allocation error when I call `vcov()` on the returned ggpredict-object.

**Comment by awcm0n** [comment]
However, I was able to compute the marginal effects with cluster-robust standard errors with `Effect()` from the _effects_ package without getting the memory allocation error. And since `ggeffect()` calls `Effect()`, it probably means there is some room for optimization.

**Comment by strengejacke** [comment]
Can you please install the latest GitHub version, and then call

``` r
vcov(result, debug = TRUE)
```

This should print the memory usage, maybe this helps narrowing down the bug.

**Comment by awcm0n** [comment]
This is what I get:
> vcov(out, debug = TRUE)
Collection 1
Garbage collection 99 = 78+7+14 (level 2) ... 
171.8 Mbytes of cons cells used (57%)
9686.0 Mbytes of vectors used (60%)
             used   (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells    3215876  171.8    5621857   300.3    5621857   300.3
Vcells 1269560883 9686.0 2117495153 16155.3 1452377023 11080.8
Collection 2
Garbage collection 102 = 80+7+15 (level 2) ... 
171.8 Mbytes of cons cells used (57%)
11879.8 Mbytes of vectors used (61%)
             used    (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells    3215938   171.8    5621857   300.3    5621857   300.3
Vcells 1557100807 11879.8 2541074183 19386.9 2108487335 16086.5
Error: cannot allocate vector of size 206.0 Gb
In addition: Warning message:
Model matrix is rank deficient. Some variance-covariance parameters are missing.
